src/sonic-eventd/rsyslog_plugin/syslog_parser.cpp

```diff
@@ -10,19 +11,59 @@
  *
 */
 
-bool SyslogParser::parseMessage(string message, string& eventTag, event_params_t& paramMap) {
+bool SyslogParser::parseMessage(string message, string& eventTag, event_params_t& paramMap, lua_State* luaState) {
     for(long unsigned int i = 0; i < m_regexList.size(); i++) {
         smatch matchResults;
         vector<string> params = m_regexList[i]["params"];
-        if(!regex_search(message, matchResults, m_expressions[i]) || params.size() != matchResults.size() - 1) {
+        if(!regex_search(message, matchResults, m_expressions[i]) || params.size() != matchResults.size() - 1 || matchResults.size() < 4) {
             continue;
         }
+
+        if(!matchResults[1].str().empty() && !matchResults[2].str().empty() && !matchResults[3].str().empty()) { // found timestamp components
+            string formattedTimestamp = m_timestampFormatter->changeTimestampFormat({ matchResults[1].str(), matchResults[2].str(), matchResults[3].str() });
+	    if(!formattedTimestamp.empty()) {
+                paramMap["timestamp"] = formattedTimestamp;
+	    } else {
+                SWSS_LOG_ERROR("Timestamp is invalid and is not able to be formatted");
+	    }
+	}
         // found matching regex
         eventTag = m_regexList[i]["tag"];
-        transform(params.begin(), params.end(), matchResults.begin() + 1, inserter(paramMap, paramMap.end()), [](string a, string b) {
-            return make_pair(a,b);
-        });
+	// check params for lua code	
+	for(long unsigned int j = 3; j < params.size(); j++) {
+            auto delimPos = params[j].find(':');
+	    string resultValue = matchResults[j + 1].str();
+	    if(delimPos == string::npos) { // no lua code
```

Comment:
Can we pre-parse this also as 
struct {
    string param_name;
    string lua_code;
} param_list[]

You may simply check "if params[i].lua_code.empty()) { ... } else { use lua code}